cob/cob.c

```diff
@@ -699,36 +896,119 @@ int m0_cob_domain_destroy(struct m0_cob_domain *dom,
 #define MKFS_ROOT_BLKSIZE       4096
 #define MKFS_ROOT_BLOCKS        16
 
-static int cob_table_delete(struct m0_be_btree *tree, struct m0_be_tx *tx,
-			     struct m0_buf *key)
+static int cob_table_delete(struct m0_btree *tree, struct m0_be_tx *tx,
+			    struct m0_buf *key)
 {
-	return M0_BE_OP_SYNC_RET(op, m0_be_btree_delete(tree, tx, &op, key),
-	                         bo_u.u_btree.t_rc);
+	struct m0_btree_op   kv_op        = {};
+	void                *k_ptr = key->b_addr;
+	m0_bcount_t          ksize = key->b_nob;
+	struct m0_btree_key  r_key = {
+				  .k_data  = M0_BUFVEC_INIT_BUF(&k_ptr, &ksize),
+				};
+
+	return M0_BTREE_OP_SYNC_WITH_RC(&kv_op,
+					m0_btree_del(tree, &r_key, NULL,
+							&kv_op, tx));
 }
 
 
-static int cob_table_update(struct m0_be_btree *tree, struct m0_be_tx *tx,
+static int btree_update_callback(struct m0_btree_cb  *cb,
+				 struct m0_btree_rec *rec)
+{
+	struct m0_btree_rec     *datum = cb->c_datum;
+
+	/** Only update the Value to the location indicated in rec. */
+	m0_bufvec_copy(&rec->r_val, &datum->r_val,
+		       m0_vec_count(&datum->r_val.ov_vec));
+	return 0;
+}
+
+static int cob_table_update(struct m0_btree *tree, struct m0_be_tx *tx,
 			     struct m0_buf *key, struct m0_buf *val)
 {
-	return M0_BE_OP_SYNC_RET(op,
-				 m0_be_btree_update(tree, tx, &op, key, val),
-	                         bo_u.u_btree.t_rc);
+	struct m0_btree_op   kv_op        = {};
+	void                *k_ptr = key->b_addr;
+	void                *v_ptr = val->b_addr;
+	m0_bcount_t          ksize = key->b_nob;
+	m0_bcount_t          vsize = val->b_nob;
+	struct m0_btree_rec  rec = {
+			   .r_key.k_data = M0_BUFVEC_INIT_BUF(&k_ptr, &ksize),
+			   .r_val        = M0_BUFVEC_INIT_BUF( &v_ptr, &vsize),
+			};
+	struct m0_btree_cb   ut_put_cb = {.c_act = btree_update_callback,
+					  .c_datum = &rec,
+					  };
+
+	return M0_BTREE_OP_SYNC_WITH_RC(&kv_op,
+					m0_btree_update(tree, &rec, &ut_put_cb,
+							&kv_op, tx));
+}
+
+static int btree_insert_callback(struct m0_btree_cb  *cb,
```

Comment:
Balloc code is using btree_ prefix for function names. Can you please change the callback names to have prefix with cob_ (for eg. cob_insert_callback).